packages/block-editor/src/store/reducer.js

```diff
@@ -239,6 +224,37 @@ function withPersistentBlockChange( reducer ) {
 	};
 }
 
+/**
+ * Higher-order reducer intended to augment the blocks reducer, assigning an
+ * `isIgnoredChange` property value corresponding to whether a change in state
+ * can be considered as ignored. A change is considered ignored when the result
+ * of an action not incurred by direct user interaction.
+ *
+ * @param {Function} reducer Original reducer function.
+ *
+ * @return {Function} Enhanced reducer function.
+ */
+function withIgnoredBlockChange( reducer ) {
+	/**
+	 * Set of action types for which a blocks state change should be ignored.
+	 *
+	 * @type {Set}
+	 */
+	const IGNORED_ACTION_TYPES = new Set( [
+		'RECEIVE_BLOCKS',
+	] );
```

Comment:
so for now is this just going to be a hardcoded list?  Would there be value in being able to customize what action types are ignored? Eg. being able to call it like this:

```js
withIgnoredBlockChange( [ 'RECEIVE_BLOCKS' ] )
```

If so, and that's useful, then the name of the HOC could maybe be changed to `withIgnoredActionChange`?

Reply:
Re `'RECEIVE_BLOCKS`  being ignored, does that mean we expect change to be ignored for any block addition to the editor (and I'm assuming changes would get picked up when there is content added to the block?)

Reply:
> Re `'RECEIVE_BLOCKS` being ignored, does that mean we expect change to be ignored for any block addition to the editor (and I'm assuming changes would get picked up when there is content added to the block?)

So, considering all of the above: Our use of `RECEIVE_BLOCKS` is and always has been exclusively a way to get blocks data for reusable blocks into state so they can be treated the same as any other content block by selectors such as `getBlock`, etc. But they're not part of the actual content. Regular user workflows to add blocks will go through `INSERT_BLOCKS` and friends. 

Long-term then: `RECEIVE_BLOCKS` shouldn't be ignored, this list won't be necessary. We probably don't need to optimize too much for it now. In fact, `RECEIVE_BLOCKS` could probably be _removed_ altogether as part of #7119, if not for compatibility's sake. Once reusable blocks are refactored as an embedded editor, we'd probably remove all of this ignoring behavior, and treat it as a real change that should trigger change detection.

In the meantime though, this basically restores the behavior we had prior to #13088 rather faithfully (including all imperfections therein).